kafka-connect/src/main/java/io/tabular/iceberg/connect/data/IcebergWriterFactory.java

```diff
@@ -86,11 +89,18 @@ private Table autoCreateTable(String tableName, SinkRecord sample) {
       spec = PartitionSpec.unpartitioned();
     }
 
-    try {
-      return catalog.createTable(identifier, schema, spec);
-    } catch (AlreadyExistsException e) {
-      LOG.info("Table {} was already created", identifier);
-      return catalog.loadTable(identifier);
-    }
+    PartitionSpec partitionSpec = spec;
+    AtomicReference<Table> result = new AtomicReference<>();
```

Comment:
Do you need to use atomic reference here? Without setting and executor it should run in the current thread. I also wonder if this could be rewritten with on failure.

Reply:
It's in a lambda so it needs to be effectively final.